// Proto file representing the EDB Term instance class

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";
import "refs.proto";

service TerminalInstanceService {
  rpc Create (TermInstCreationMessage) returns (EDBObjMessage) {}

  rpc GetOwningCellInstance (EDBObjMessage) returns (EDBObjMessage) {}

  rpc GetDefinitionTerminal (EDBObjMessage) returns (EDBObjMessage) {}

  rpc GetDefinitionTerminalName (EDBObjMessage) returns (google.protobuf.StringValue) {}
}

message TermInstCreationMessage {
  EDBObjMessage layout = 1;
  NetRefMessage net = 2;
  EDBObjMessage cell_inst = 3;
  string name = 4;
}
